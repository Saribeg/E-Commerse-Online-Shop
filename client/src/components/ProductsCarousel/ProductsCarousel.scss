@import "../../scss/reset.scss";
@import "../../scss/resources/resources.scss";

.products-carousel {
  margin: 0 auto;
  text-align: center;
}

.carousel .slide {
  background: #fff;
}

.slide.selected {
  position: relative;
}

.carousel-item {
  img {
    height: auto;
  }

  .price {
    position: absolute;
    left: 30px;
    top: 30px;
    color: $black;
    font-size: 24px;
    border: 1px solid $black;
    border-radius: 500px;
    padding: 40px 10px;
  }
}

.carousel .slide {

  .info {
    letter-spacing: 2px;
    color: $black;
    font-size: 40px;
    font-weight: bold;
    position: absolute;
    top: 25%;
    left: 70%;
    width: 30%;
    text-align: left;
  }

  .carousel-btn {
    position: absolute;
    top: 50%;
    left: 70%;
    width: 200px;
    height: 40px;
    border-radius: 100px;
    border: 1px solid $black;
    background: transparent;
    letter-spacing: 1.4px;
    font-size: 18px;
    color: $black;

    &:hover {
      cursor: pointer;
      color: $red;
      border-color: $red;
      transition: all 0.3s ease;
    }
  }
}

