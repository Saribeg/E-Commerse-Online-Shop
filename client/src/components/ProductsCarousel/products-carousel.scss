@import "../../scss/resources/resources.scss";

.products-carousel {
  margin: 0 auto;
  text-align: center;

  @include max-width($landscape-tablet) {
    margin-top: 49px;
  }

  @include max-width($landscape-mobile-big) {
    display: none;
  }
}

.slide.selected {
  position: relative;
}

.carousel-item {
  position: relative;

  img {
    height: auto;
  }

  .price {
    position: absolute;
    left: 50px;
    top: 72px;
    color: $black;
    font-size: 20px;

    &:before {
      border-bottom: 2px solid $black;
      position: absolute;
      content: "";
      width: 100%;
      height: 50%;
      transform: rotate(-7deg);
    }

    @include max-width($landscape-tablet) {
      left: 30px;
      top: 20px;
      font-size: 18px;
      padding: 10px;
    }

    @include max-width($landscape-mobile-big) {
      left: 20px;
      top: 20px;
      font-size: 14px;
      padding: 5px;
    }
  }

  .stock-price {
    position: absolute;
    left: 130px;
    top: 50px;
    color: $black;
    font-size: 24px;
    border: 2px solid $black;
    border-radius: 100px;
    padding: 20px;
    font-weight: 600;

    @include max-width($landscape-tablet) {
      left: 30px;
      top: 50px;
      font-size: 20px;
      padding: 10px;
    }

    @include max-width($landscape-mobile-big) {
      left: 40px;
      top: 15px;
      font-size: 16px;
      padding: 5px;
    }
  }
}

.carousel .slide {
  background: #fff;

  .info {
    letter-spacing: 2px;
    color: $black;
    font-size: 40px;
    font-weight: bold;
    position: absolute;
    top: 25%;
    left: 70%;
    width: 30%;
    text-align: left;

    @include max-width($landscape-tablet) {
      font-size: 20px;
    }

    @include max-width($landscape-mobile-big) {
      font-size: 12px;
    }
  }

  .carousel-btn {
    position: absolute;
    top: 50%;
    left: 70%;
    width: 200px;
    height: 40px;
    border-radius: 100px;
    border: 1px solid $black;
    line-height: 40px;
    background: transparent;
    letter-spacing: 1.4px;
    font-size: 18px;
    color: $black;
    outline: none;

    &:hover {
      cursor: pointer;
      color: $red;
      border-color: $red;
      transition: all 0.3s ease;
    }

    @include max-width($landscape-tablet) {
      top: 60%;
      width: 120px;
      height: 30px;
      font-size: 14px;
      line-height: 30px;
    }

    @include max-width($landscape-mobile-big) {
      width: 80px;
      height: 20px;
      font-size: 10px;
      line-height: 20px;
    }
  }
}

