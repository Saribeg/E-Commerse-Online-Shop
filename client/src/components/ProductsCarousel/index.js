import React, { Component } from "react";
import "react-responsive-carousel/lib/styles/carousel.min.css";
import { Carousel } from "react-responsive-carousel";
import "./products-carousel.scss";
import { NavLink } from "react-router-dom";
import axios from "axios";

class ProductsCarousel extends Component {
  state = {
    dbCarousel: []
  };

  componentDidMount() {
    axios.get(`/carousel`).then(res => {
      this.setState({ dbCarousel: res.data });
    });
  }

  render() {
    let itemsCarousel = this.state.dbCarousel.map(item => {
      return (
        <div id={item.desc} key={item._id} className="carousel-item">
          <img src={item.imageUrl} alt="product" />
          <div className="price">${item.product.previousPrice}</div>
          <div className="stock-price">${item.product.currentPrice}</div>
          <div className="info">{item.product.model}</div>
          <NavLink
            key={item.product._id}
            to={item.product.productUrl}
            className="carousel-btn"
            onClick={() => (this.setState.dbCarousel = [])}
          >
            Show more
          </NavLink>
        </div>
      );
    });

    return (
      <>
        {itemsCarousel.length < 1 ? null : (
          <Carousel
            className="products-carousel"
            // autoPlay={true}
            interval={5000}
            transitionTime={700}
            stopOnHover={true}
            showIndicators={false}
            infiniteLoop={true}
            showStatus={false}
            useKeyboardArrows={true}
          >
            {itemsCarousel}
          </Carousel>
        )}
      </>
    );
  }
}

export default ProductsCarousel;
