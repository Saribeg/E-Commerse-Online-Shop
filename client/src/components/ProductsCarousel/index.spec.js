import React from 'react';
import ProductsCarousel from './index';

describe('ProductsCarousel component', () => {
    it('ProductsCarousel component render', () => {
        const wrapper  = shallow(<ProductsCarousel />)
        expect(wrapper).toMatchSnapshot()
    })
})