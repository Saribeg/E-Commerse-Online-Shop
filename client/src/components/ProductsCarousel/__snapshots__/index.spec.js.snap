// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsCarousel component ProductsCarousel component render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductsCarousel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoPlay": true,
      "axis": "horizontal",
      "centerMode": false,
      "centerSlidePercentage": 80,
      "children": Array [],
      "className": "products-carousel",
      "dynamicHeight": false,
      "emulateTouch": false,
      "infiniteLoop": true,
      "interval": 5000,
      "onChange": [Function],
      "onClickItem": [Function],
      "onClickThumb": [Function],
      "selectedItem": 0,
      "showArrows": true,
      "showIndicators": false,
      "showStatus": false,
      "showThumbs": true,
      "statusFormatter": [Function],
      "stopOnHover": true,
      "swipeScrollTolerance": 5,
      "swipeable": true,
      "transitionTime": 700,
      "useKeyboardArrows": true,
      "verticalSwipe": "standard",
      "width": "100%",
    },
    "ref": null,
    "rendered": Array [],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoPlay": true,
        "axis": "horizontal",
        "centerMode": false,
        "centerSlidePercentage": 80,
        "children": Array [],
        "className": "products-carousel",
        "dynamicHeight": false,
        "emulateTouch": false,
        "infiniteLoop": true,
        "interval": 5000,
        "onChange": [Function],
        "onClickItem": [Function],
        "onClickThumb": [Function],
        "selectedItem": 0,
        "showArrows": true,
        "showIndicators": false,
        "showStatus": false,
        "showThumbs": true,
        "statusFormatter": [Function],
        "stopOnHover": true,
        "swipeScrollTolerance": 5,
        "swipeable": true,
        "transitionTime": 700,
        "useKeyboardArrows": true,
        "verticalSwipe": "standard",
        "width": "100%",
      },
      "ref": null,
      "rendered": Array [],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
